client/src/components/Filedrop/index.js

```diff
@@ -0,0 +1,43 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import Dropzone from 'react-dropzone';
+import classNames from 'classnames';
+import './Filedrop.css';
+
+export default class Filedrop extends Component {
+    static propTypes = {
+        accepted: PropTypes.instanceOf(File),
+        rejected: PropTypes.instanceOf(File),
+        onAccept: PropTypes.func.isRequired,
+        onReject: PropTypes.func.isRequired,
+    }
+
+    filedrop = (accepted, rejected) => {
+        const { onAccept, onReject } = this.props;
+        if (rejected.length) {
+            onReject(rejected[0]);
+            return;
+        }
+        onAccept(accepted[0]);
+    }
+
+    render() {
+        const { accepted, rejected } = this.props;
+        return (
+                <Dropzone className={ classNames('filedrop', { accepted, rejected }) } activeClassName="active" maxSize={ 10000 } multiple={ false } onDrop={ this.filedrop }>
```

Comment:
Can you also add a condition that the file has a file extension. Whether that's here or in the submit action. On the server, I require a supported file extension(currently it's only `.java` or `.py`).

We could also potentially think about adding allowed languages on each problem/assignment/course.